CORS fails for path with period and space

## Environment Details
* Helidon Version: 4.x - main

----------

## Problem Description
CORS Aggregator operates with unnormalized request path while using `UriPathNoParam::decode`
which uses `URI.create` for paths containing period.
```java
        if (!validate && percent == -1 && doubleSlash == -1 && dot == -1) {
            return rawPath;
        }
        ...
        return URI.create(rawPath).normalize().getPath();
```

`URI.create` then fails for paths like `/v1.0/ui/All Account List View/SIS Account List Applet` with:
```
Illegal character in path at index 12: /v1.0/ui/All Account List View/SIS Account List Applet
```

## Steps to reproduce
```diff
diff --git a/cors/src/test/java/io/helidon/cors/AggregatorTest.java b/cors/src/test/java/io/helidon/cors/AggregatorTest.java
--- a/cors/src/test/java/io/helidon/cors/AggregatorTest.java	(revision ba7345f4735db7398e5c0b15eed47bb48aa5209a)
+++ b/cors/src/test/java/io/helidon/cors/AggregatorTest.java	(date 1743089828770)
@@ -57,6 +57,8 @@
 
     @Test
     void testGreet() {
+        checkMatch("/openintegration/v10/ui/All Account List View/SIS Account List Applet", "GET", 3);
+        checkMatch("/openintegration/v1.0/ui/All Account List View/SIS Account List Applet", "GET", 3);
         checkMatch("/greet", "GET", 2);
         checkMatch("/greet", "DELETE", 1);
         checkMatch("/greet", "HEAD", 3);

```

Whole stacktrace:
```
2025.03.27 16:44:27.277 WARNING Internal server error
io.helidon.http.RequestException: Illegal character in path at index 12: /v1.0/ui/All Account List View/SIS Account List Applet
	at io.helidon.http.RequestException$Builder.build(RequestException.java:139)
	at io.helidon.webserver.http.ErrorHandlers.unhandledError(ErrorHandlers.java:205)
	at io.helidon.webserver.http.ErrorHandlers.lambda$handleError$1(ErrorHandlers.java:184)
	at java.base/java.util.Optional.ifPresentOrElse(Optional.java:198)
	at io.helidon.webserver.http.ErrorHandlers.handleError(ErrorHandlers.java:183)
	at io.helidon.webserver.http.ErrorHandlers.runWithErrorHandling(ErrorHandlers.java:119)
	at io.helidon.webserver.http.Filters.filter(Filters.java:77)
	at io.helidon.webserver.http.HttpRoutingImpl.route(HttpRoutingImpl.java:73)
	at io.helidon.webserver.http1.Http1Connection.route(Http1Connection.java:544)
	at io.helidon.webserver.http1.Http1Connection.handle(Http1Connection.java:223)
	at io.helidon.webserver.ConnectionHandler.run(ConnectionHandler.java:169)
	at io.helidon.common.task.InterruptableTask.call(InterruptableTask.java:47)
	at io.helidon.webserver.ThreadPerTaskExecutor$ThreadBoundFuture.run(ThreadPerTaskExecutor.java:239)
	at java.base/java.lang.VirtualThread.run(VirtualThread.java:329)
Caused by: java.lang.IllegalArgumentException: Illegal character in path at index 12: /v1.0/ui/All Account List View/SIS Account List Applet
	at java.base/java.net.URI.create(URI.java:932)
	at io.helidon.common.uri.UriPathNoParam.decode(UriPathNoParam.java:136)
	at io.helidon.common.uri.UriPathNoParam.path(UriPathNoParam.java:76)
	at io.helidon.http.PathMatchers$PatternPathMatcher.match(PathMatchers.java:419)
	at io.helidon.cors.Aggregator$CrossOriginConfigMatchable.matches(Aggregator.java:330)
	at io.helidon.cors.Aggregator.lambda$findFirst$0(Aggregator.java:299)
	at java.base/java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:193)
	at java.base/java.util.stream.ReferencePipeline$15$1.accept(ReferencePipeline.java:556)
	at java.base/java.util.ArrayList$ArrayListSpliterator.tryAdvance(ArrayList.java:1686)
	at java.base/java.util.stream.ReferencePipeline.forEachWithCancel(ReferencePipeline.java:144)
	at java.base/java.util.stream.AbstractPipeline.copyIntoWithCancel(AbstractPipeline.java:574)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:560)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:546)
	at java.base/java.util.stream.FindOps$FindOp.evaluateSequential(FindOps.java:150)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:265)
	at java.base/java.util.stream.ReferencePipeline.findFirst(ReferencePipeline.java:662)
	at io.helidon.cors.Aggregator.findFirst(Aggregator.java:304)
	at io.helidon.cors.Aggregator.lookupCrossOrigin(Aggregator.java:280)
	at io.helidon.cors.CorsSupportHelper.processCorsRequest(CorsSupportHelper.java:477)
	at io.helidon.cors.CorsSupportHelper.processRequest(CorsSupportHelper.java:303)
	at io.helidon.webserver.cors.CorsSupport.handle(CorsSupport.java:97)
	at io.helidon.webserver.http.HttpRoutingImpl$RoutingExecutor.doRoute(HttpRoutingImpl.java:165)
	at io.helidon.webserver.http.HttpRoutingImpl$RoutingExecutor.call(HttpRoutingImpl.java:124)
	at io.helidon.webserver.http.HttpRoutingImpl$RoutingExecutor.call(HttpRoutingImpl.java:102)
	at io.helidon.webserver.http.ErrorHandlers.runWithErrorHandling(ErrorHandlers.java:76)
	... 8 more
Caused by: java.net.URISyntaxException: Illegal character in path at index 12: /v1.0/ui/All Account List View/SIS Account List Applet
	at java.base/java.net.URI$Parser.fail(URI.java:2995)
	at java.base/java.net.URI$Parser.checkChars(URI.java:3166)
	at java.base/java.net.URI$Parser.parseHierarchical(URI.java:3248)
	at java.base/java.net.URI$Parser.parse(URI.java:3207)
	at java.base/java.net.URI.<init>(URI.java:645)
	at java.base/java.net.URI.create(URI.java:930)
	... 32 more
```

--- a/cors/src/main/java/io/helidon/cors/Aggregator.java
+++ b/cors/src/main/java/io/helidon/cors/Aggregator.java
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2020, 2024 Oracle and/or its affiliates.
+ * Copyright (c) 2020, 2025 Oracle and/or its affiliates.
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -23,6 +23,7 @@ import java.util.function.Supplier;
 
 import io.helidon.common.config.Config;
 import io.helidon.common.config.ConfigValue;
+import io.helidon.common.uri.UriEncoding;
 import io.helidon.common.uri.UriPath;
 import io.helidon.cors.LogHelper.MatcherChecks;
 import io.helidon.http.PathMatcher;
@@ -327,7 +328,8 @@ public class Aggregator {
         }
 
         boolean matches(String path, String method) {
-            return matcher.match(UriPath.create(path)).accepted() && get().matches(method);
+            return matcher.match(UriPath.create(UriEncoding.encode(path, UriEncoding.Type.PATH))).accepted()
+                    && get().matches(method);
         }
 
         PathMatcher matcher() {

--- a/cors/src/test/java/io/helidon/cors/AggregatorTest.java
+++ b/cors/src/test/java/io/helidon/cors/AggregatorTest.java
@@ -1,5 +1,5 @@
 /*
- * Copyright (c) 2020, 2022 Oracle and/or its affiliates.
+ * Copyright (c) 2020, 2025 Oracle and/or its affiliates.
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
@@ -42,6 +42,14 @@ class AggregatorTest {
         builder.addCrossOrigin("/greet", addToMap(2, CrossOriginConfig.builder()
                 .allowMethods("GET")
                 .build()));
+        builder.addCrossOrigin("/openintegration/v10/ui/space in path",
+                               addToMap(6, CrossOriginConfig.builder()
+                                       .allowMethods("GET", "PUT")
+                                       .build()));
+        builder.addCrossOrigin("/openintegration/v1.0/ui/dotInPath",
+                               addToMap(7, CrossOriginConfig.builder()
+                                       .allowMethods("GET", "PUT")
+                                       .build()));
         builder.addPathlessCrossOrigin(addToMap(3, CrossOriginConfig.builder()
                 .allowMethods("GET", "HEAD", "OPTIONS")
                 .build()));
@@ -51,6 +59,7 @@ class AggregatorTest {
         builder.addCrossOrigin("/greet/{+}", addToMap(5, CrossOriginConfig.builder()
                 .allowMethods("POST")
                 .build()));
+
 
         aggregator = builder.build();
     }
@@ -82,6 +91,19 @@ class AggregatorTest {
     void testWildcardedPath() {
         checkNoMatch("/greet", "POST");
         checkMatch("/greet/sub", "POST", 5);
+    }
+
+    @Test
+    void testSpecialCharacters() {
+        /*
+        The first two checks should match the entries added for those two specific paths (IDs 6 and 7 respectively).
+        The second two checks make sure that wildcarded matches--the "pathless" entry ID 3--work with paths containing
+        dots or spaces.
+         */
+        checkMatch("/openintegration/v10/ui/space in path", "GET", 6);
+        checkMatch("/openintegration/v1.0/ui/dotInPath", "GET", 7);
+        checkMatch("/openintegration/v20/ui/space in path", "GET", 3);
+        checkMatch("/openintegration/v2.0/ui/dot and space in path", "GET", 3);
     }
 
     private static CrossOriginConfig addToMap(int ID, CrossOriginConfig coc) {

--- a/cors/src/test/java/io/helidon/cors/AggregatorTest.java
+++ b/cors/src/test/java/io/helidon/cors/AggregatorTest.java
@@ -50,6 +50,11 @@ class AggregatorTest {
                                addToMap(7, CrossOriginConfig.builder()
                                        .allowMethods("GET", "PUT")
                                        .build()));
+        builder.addCrossOrigin("/openintegration/{+}",
+                               addToMap(8, CrossOriginConfig.builder()
+                                       .allowMethods("GET", "PUT")
+                                       .build()));
+
         builder.addPathlessCrossOrigin(addToMap(3, CrossOriginConfig.builder()
                 .allowMethods("GET", "HEAD", "OPTIONS")
                 .build()));
@@ -97,13 +102,12 @@ class AggregatorTest {
     void testSpecialCharacters() {
         /*
         The first two checks should match the entries added for those two specific paths (IDs 6 and 7 respectively).
-        The second two checks make sure that wildcarded matches--the "pathless" entry ID 3--work with paths containing
-        dots or spaces.
+        The second two checks make sure that wildcarded matches work with paths containing dots or spaces.
          */
         checkMatch("/openintegration/v10/ui/space in path", "GET", 6);
         checkMatch("/openintegration/v1.0/ui/dotInPath", "GET", 7);
-        checkMatch("/openintegration/v20/ui/space in path", "GET", 3);
-        checkMatch("/openintegration/v2.0/ui/dot and space in path", "GET", 3);
+        checkMatch("/openintegration/v20/ui/space in path", "GET", 8);
+        checkMatch("/openintegration/v2.0/ui/dot and space in path", "GET", 8);
     }
 
     private static CrossOriginConfig addToMap(int ID, CrossOriginConfig coc) {

